src/utils/getUrlByRequestOptions.ts

```diff
@@ -166,7 +136,7 @@ export function getUrlByRequestOptions(options: ResolvedRequestOptions): URL {
     : ''
   logger.info('auth string:', authString)
 
-  const url = new URL(`${protocol}//${hostname}${path}`)
+  const url = new URL(`${protocol}//${host}${path}`)
```

Comment:
You are correct regarding this change though. We should use `host` here, not `hostname`.